packages/dds/tree/src/simple-tree/core/treeNodeKernel.ts

```diff
@@ -80,37 +122,50 @@ export class TreeNodeKernel implements Listenable<TreeChangeEvents> {
 		treeNodeToKernel.set(node, this);
 	}
 
+	/**
+	 * Transition from {@link Unhydrated} to hydrated.
+	 * @remarks
+	 * Happens at most once for any given node.
+	 */
 	public hydrate(anchorNode: AnchorNode): void {
-		const offChildrenChanged = anchorNode.on("childrenChangedAfterBatch", () => {
-			this.#events.emit("nodeChanged");
-		});
-
-		const offSubtreeChanged = anchorNode.on("subtreeChangedAfterBatch", () => {
-			this.#events.emit("treeChanged");
-		});
-
-		const offAfterDestroy = anchorNode.on("afterDestroy", () => this.dispose());
+		assert(!this.disposed, "cannot use a disposed node");
 
 		this.#hydrated = {
 			anchorNode,
-			offAnchorNode: () => {
-				offChildrenChanged();
-				offSubtreeChanged();
-				offAfterDestroy();
-			},
+			offAnchorNode: new Set([
+				anchorNode.on("afterDestroy", () => this.dispose()),
+				// TODO: this should be triggered on change even for unhydrated nodes.
```

Comment:
Does this need to be done now? What problems are caused by the current situation?

Reply:
you will not get any errors if you iterate unhydrated nodes while mutating them. I think thats thew only thing we currently use generation numbers for.

Reply:
note that this is a preexisting issue unrelated to this PR which is simply being documented.